api/api.go

```diff
@@ -241,17 +237,24 @@ func (api *Wrapper) CreateAccessToken(ctx echo.Context) (err error) {
 		errorResponse := AccessTokenRequestFailedResponse{Error: "invalid_grant", ErrorDescription: errDesc}
 		return ctx.JSON(http.StatusBadRequest, errorResponse)
 	}
-	vendorID := vendorIdentifierFromHeader(ctx)
-	if vendorID == "" {
-		errDesc := "Vendor identifier missing in header"
-		errorResponse := AccessTokenRequestFailedResponse{Error: "invalid_grant", ErrorDescription: errDesc}
+
+	if params.XSslClientCert == "" {
+		errDesc := "Client certificate missing in header"
+		errorResponse := AccessTokenRequestFailedResponse{Error: "invalid_request", ErrorDescription: errDesc}
```

Comment:
aren't there consts for these OAUTH spec error codes? (if not, maybe we should make 'm)

Reply:
added constants to api layer